src/cbc_sdk/platform/devices.py

```diff
@@ -315,6 +318,124 @@ def nsx_remediation(self, tag, set_tag=True):
             return None  # clearing tag is a no-op in this case
         return NSXRemediationJob.start_request(self._cb, self.id, tag, set_tag)
 
+    def get_asset_group_ids(self, **kwargs):
+        """
+        Finds the list of asset group IDs that this device is a member of.
+
+        Required Permissions:
+            group-management(READ)
+
+        Parameters:
+            **kwargs (dict): Keyword arguments as documented below.
+
+        Keyword Args:
+            filter (str): Can restrict the types of group membership returned by this method.  Values are "ALL"
+                          to return all groups, "DYNAMIC" to return only groups that each member belongs to via the
+                          asset group query, or "MANUAL" to return only groups that the members were manually added to.
+                          Default is "ALL".
+
+        Returns:
+            list[str]: A list of asset group IDs this device belongs to.
+        """
+        rc = Device.get_asset_groups_for_devices(self._cb, self, **kwargs)
+        return rc[str(self._model_unique_id)]
+
+    def get_asset_groups(self, **kwargs):
+        """
+        Finds the list of asset groups that this device is a member of.
+
+        Required Permissions:
+            group-management(READ)
+
+        Parameters:
+            **kwargs (dict): Keyword arguments as documented below.
+
+        Keyword Args:
+            filter (str): Can restrict the types of group membership returned by this method.  Values are "ALL"
+                          to return all groups, "DYNAMIC" to return only groups that each member belongs to via the
+                          asset group query, or "MANUAL" to return only groups that the members were manually added to.
+                          Default is "ALL".
+
+        Returns:
+            list[AssetGroup]: A list of asset groups this device belongs to.
+        """
+        rc = Device.get_asset_groups_for_devices(self._cb, self, **kwargs)
+        return [self._cb.select("AssetGroup", v) for v in rc[str(self._model_unique_id)]]
+
+    @staticmethod
+    def _normalize_member_list(member_list):
```

Comment:
Should this be a more generic helper function in the CBC SDK to normalize a list and you can specify the key to pull out of an object?